packages/ngtools/webpack/src/ngcc_processor.ts

```diff
@@ -54,6 +55,46 @@ export class NgccProcessor {
       return;
     }
 
+    // Perform a ngcc run check to determine if an initial execution is required.
+    // If a run hash file exists that matches the current package manager lock file and the
+    // project's tsconfig, then an initial ngcc run has already been performed.
+    let skipProcessing = false;
+    let runHashFilePath: string | undefined;
+    const ngccPath = path.join(this._nodeModulesDirectory, '.ngcc');
+    try {
+      let lockData;
+      let lockFile = 'yarn.lock';
+      try {
+        lockData = readFileSync(path.join(this._nodeModulesDirectory, '..', lockFile));
+      } catch {
+        lockFile = 'package-lock.json';
+        lockData = readFileSync(path.join(this._nodeModulesDirectory, '..', lockFile));
+      }
+      const relativeTsconfigPath = path.relative(this.basePath, this.tsConfigPath);
+      const tsconfigData = readFileSync(this.tsConfigPath);
+
+      // Generate a hash that represents the state of the package lock file and used tsconfig
+      const runHash = createHash('sha256')
+        .update(lockData)
+        .update(lockFile)
+        .update(tsconfigData)
+        .update(relativeTsconfigPath)
+        .digest('hex');
+
+      runHashFilePath = path.join(ngccPath, runHash + '.lock');
```

Comment:
Nice idea to use the hash in the filename, which prevents race conditions on reading the file.